src/intellisense/langIntellisenseProvider.ts

```diff
@@ -646,21 +646,43 @@ class ReferenceCompletionProvider extends IntellisenseProvider implements vscode
                     return labels.map(label => new vscode.CompletionItem(label, vscode.CompletionItemKind.Reference));
                 }).flat();
             case 1:
-                const bibUri = vfs.pathToUri(`${OUTPUT_FOLDER_NAME}/output.bbl`);
-                const content = new TextDecoder().decode( await vfs.openFile(bibUri) );
-                const regex = /\\bibitem\{([^\}]*)\}/g;
                 const items = new Array<vscode.CompletionItem>();
-                let match: RegExpExecArray | null;
-                while (match = regex.exec(content)) {
-                    const item = new vscode.CompletionItem(match[1], vscode.CompletionItemKind.Reference);
-                    items.push(item);
+                await this.getBibCompletionItems(vfs, '/', items);
```

Comment:
so confusing, stop using copilot please.

Since reference passing is c-style, please return and concat the items

Reply:
Now it use `.then({})` to push items back. Though, ... this sentence is not written by copilot...😇

Reply:
nice, but don't use `then` promise cause you are already using `await`.

Reply:
True, now it directly push the return value back.